### PR TITLE
Handle more buildplanner errors

### PR DESCRIPTION
**Pull request by MDrakos**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-2821" title="DX-2821" target="_blank">TaskDX-2821</a>  Handle Buildplanner errors with user facing messages
  </td></table>
  <br />

### CODE REVIEW

File: pkg/platform/api/buildplanner/response/commit.go [hunk 1]
```diff
@@ -133,9 +133,9 @@ type TargetNotFound struct {
 	PossibleTargets []string `json:"possibleTargets"`
 }
 
-type ValidationError struct {
-	SubErrors []*BuildExprError `json:"subErrors"`
-}
+// ValidationError represents a validation error that occurred during planning.
+// Contains message and subErrors which is handled separately
+type ValidationError struct{}
```
Comment (MDrakos) [inline pkg/platform/api/buildplanner/response/commit.go:138]: We now have multiple errors with `SubErrors` so I moved that to the common `Error` struct

**Comment by MDrakos** [comment]
I found that a lot of the errors we are currently handling wouldn't be very enriched by the extra fields. I also took the time in this PR to handle errors that we weren't before.

**Review by Naatan — CHANGES_REQUESTED** [review]
We should be using the rationalizer for this. These are already returning typed errors, which are at least to some level already rationalized. Adding localized errors underneath them feels inappropriate. We should only be marking input errors outside of rationalizers if there is a clear reason why rationalizing doesn't work there.

**Comment by MDrakos** [comment]
Closing this PR and filed another story that I think will better address our buildplanner error handling.

**Pull request closed without merging**